src/expr/symbol_table.cpp

```diff
@@ -87,12 +106,43 @@ class SymbolTable::Implementation {
 
   /** A set of defined functions. */
   CDHashSet<Expr, ExprHashFunction>* d_functions;
+  //------------------------ operator overloading
+  // the null expression
+  Expr d_nullExpr;
+  // This data structure stores a trie of expressions with
+  // the same name, and must be distinguished by their argument types.
+  // For simplicity, it is context-independent.
```

Comment:
Making it context-independent seems like a potential problem in the long run.  Maybe add an issue to revisit this decision?